### PR TITLE
Support IJava kernel export runnable zip files

### PR DESCRIPTION
**Pull request by lanking520**
Now this commit allows basic

```
%export model
```
operation to do model compiling & packaging. A bunch of corner cases are not considered. Two remaining TODO items:

1. Format the gradle project and upload to somewhere
2. Setup general test cases to verify the code is working.

### CODE REVIEW

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        }
+        this.eva = evaluator;
+    }
+
+    @LineMagic(aliases = "export")
+    public void exec(List<String> args) throws IOException {
+        if (args.size() != 1) {
+            throw new RuntimeException("Require single name input");
+        }
+        Path src = Paths.get("template/gradle_project");
```
Comment (frankfliu) [inline src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java]: This folder won't exist in the jar, we can package a template.zip in the jar file.

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        System.out.println("Start gathering parameters...");
+        exportArtifact(args.get(0), dest);
+        System.out.println("Template created in: " + dest.toString());
+        writeCode(dest);
+        writeDeps(dest);
+        System.out.println("Start generation...");
+        buildJar(dest);
+        System.out.println("Build completed! Zipping everything...");
+        Path zipFile = zipContent(dest);
+        System.out.println("Finished! you can find the zip file here: " + zipFile.toAbsolutePath().toString());
```
Comment (frankfliu) [inline src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java:51]: We should save the output file in current directory or a user specified location.

File: src/main/java/io/github/spencerpark/ijava/JavaKernel.java [hunk 3]
```diff
@@ -93,12 +99,18 @@ public JavaKernel() {
                 .build();
         this.mavenResolver = new MavenResolver(this::addToClasspath);
 
+        this.recorders = new ArrayList<>();
+        this.recorders.add(new ImportRecorder());
```
Comment (frankfliu) [inline src/main/java/io/github/spencerpark/ijava/JavaKernel.java]: We should combine 3 recorders into one class, and the class will have 3 method:
1. getImportContent()
2. getClassContent()
3. getDependencyContent()
or expose 2 APIs:
1. updateBuildGradle(Path path)
2. writeClass(Path path)

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        } catch (Exception e) {
+            throw new RuntimeException("Export failed, due to", e);
+        }
+    }
+
+    private void writeCode(Path dest) {
+        try {
+            Path dir = dest.resolve("src/main/java/ai/djl/examples");
+            Files.createDirectories(dir);
+            BufferedWriter writer = Files.newBufferedWriter(dir.resolve("Export.java"));
```
Comment (frankfliu) [inline src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java:70]: You are write all classes into on file, this might causing problem in future.
We could write a top level class, and make all other classes as public static inner class

File: src/main/java/io/github/spencerpark/ijava/JavaKernel.java (hunk 3 above)
Comment (lanking520) [reply src/main/java/io/github/spencerpark/ijava/JavaKernel.java]: Since it is a role like recorder, I will not put 4 and 5 in the class. These can still in the ExportMagic

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java (hunk 1 above)
Comment (lanking520) [reply src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java]: Yeah, probably use the ZipUtils.unzip(Url.openStream(), dest) to unzip the folder

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java (hunk 2 above)
Comment (lanking520) [reply src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java:51]: Will make it to current dir

File: src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java (hunk 4 above)
Comment (lanking520) [reply src/main/java/io/github/spencerpark/ijava/magics/ExportMagics.java:70]: Do you mean something like:

```java
class Export {
   public static class A { }
   public static class B  {}
}
```
Can we find the class (e.g BlockFactory) through the searching?

**Comment by lanking520** [comment]
@frankfliu please checkout the generated file: 
[exported.zip](https://github.com/frankfliu/IJava/files/6241181/exported.zip)
